packages/driver/src/cypress/log.ts

```diff
@@ -444,7 +444,7 @@ class Log {
     this.obj = {
       highlightAttr: HIGHLIGHT_ATTR,
       numElements: $el.length,
-      visible: $el.length === $el.filter(':visible').length,
+      visible: this.get('crossOriginLog') ? this.get('visible') : $el.length === $el.filter(':visible').length,
```

Comment:
Just curious, why do we need to track if the log was a cross origin log or not?

I was looking into the code, where are we setting if this is visible?

Reply:
It's a bit strange. So `visible` is set on the original cross origin log in the spec bridge when `$el` is set in the log. When it is reified back in the primary, the element is no longer visible since the primary wouldn't have access to the current DOM to compare against, and even if it did time might have elapsed on reification where the element may not be visible any longer. The main reason for the `crossOriginLog` here is to preserve `visible` at the time of setting in the spec bridge and to make sure it does not get overwritten on reification. Do you think a comment here along those lines would go a long way?

Reply:
That's the idea. A concern I have with the suggestion is if the element was previously visible in the log, and then when the log is changed the element is no longer visible. For instance, if I did something like `cy.get('hide me').click()`, the element would have once been visible and then isn't after the `click` log changes. This is one of the reasons I have `crossOriginLog` just take the `visible` attribute at face value, however it was set in the spec bridge, because it could be `true` or `false` for the same log after a change event at any given time. Otherwise, this could have been an awesome spot for [nullish coalescing](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing)

Reply:
I mean, I see what you are saying, however from what I can tell with other commands, typically the $el is either the found element (cy.get) or the received subject (chainable command like .should) at the time the command executes. I don't recall many command that has updated the `el` attribute after the command has executed. It could be dependent on the command, But since it's a JQuery obj, to me it seem this is taking the as-is visibly of the subject at command exec time.

Reply:
> For instance, if I did something like cy.get('hide me').click(), the element would have once been visible and then isn't after the click log changes.

Also I do think .click() produced before and after screenshots of the original subject and then the dom after click.

Reply:
I can't find any that update either. I believe you are right that it is just taking the visibility as-in on the log. We definitely want to keep the visibility as-in at the time of command execution and not reification, so I am starting to think that we can leverage the coalescer here. So if `visible` is `true` or `false` or is already set, do nothing, otherwise calculate it. Then we can ditch `crossOriginlog` all together. How does that sound?

Reply:
@emilyrohrbough I gave this a shot in 6926bfe86f. I think it has the results we are looking for but let me know!